### PR TITLE
Update MS_Marco to 2.1

### PR DESCRIPTION
**Pull request by isaacmg**
Updates the URLs then adds a conversion function to convert the JSON to the original JSONL format.

### CODE REVIEW

File: parlai/tasks/ms_marco/build.py [hunk 1]
```diff
@@ -9,22 +9,34 @@
...
-    lines = [x.decode('utf-8') for x in f.readlines()]
-    if delete_gz:
-        os.remove(filename)
+def read_file(filename):
+    print("reached line loading")
+    with open(filename) as f:
+        lines = [x for x in f.readlines()]
     return lines
 
+def convert_file(input_file_path, output_file_path):
```
Comment (stephenroller) [inline parlai/tasks/ms_marco/build.py]: Yes, please get rid of the dependency on pandas. It looks like it could be replaced with roughly:

```
records = json.read(f)
#...
for row in records:
    f.write(json.dumps(row))
    f.write('\n')
```

File: parlai/tasks/ms_marco/build.py [hunk 2]
```diff
@@ -9,22 +9,34 @@
...
 import parlai.core.build_data as build_data
 
 
-def read_gz(filename, delete_gz=True):
-    f = gzip.open(filename, 'rb')
-    lines = [x.decode('utf-8') for x in f.readlines()]
-    if delete_gz:
-        os.remove(filename)
+def read_file(filename):
+    print("reached line loading")
```
Comment (stephenroller) [inline parlai/tasks/ms_marco/build.py]: Yes, please drop this println

File: parlai/tasks/ms_marco/build.py [hunk 3]
```diff
@@ -57,7 +66,7 @@ def create_fb_format(outpath, dtype, inpath):
 
 def build(opt):
     dpath = os.path.join(opt['datapath'], 'MS_MARCO')
-    version = None
+    version = None
```
Comment (stephenroller) [inline parlai/tasks/ms_marco/build.py]: This and a few other places will likely have some lint errors.

**Comment by isaacmg** [comment]
So I think the last change should get things working even though Travis is still throwing a memory error. I tested it in parts on my laptop but I'm going to attempt a full build of the dataset from scratch in the next few days on GCP just to confirm. Besides that I think it's just the lint errors like you said. If you have any suggestions to resolve the memory error please let me.

**Comment by isaacmg** [comment]
Yeah I don't know how that file got added just removed it though. I'm working on fixing the last couple lint errors.

**Comment by stephenroller** [comment]
There's a weird issue with `test_new_data` being broken and I don't understand why.

**Comment by stephenroller** [comment]
Critical test passes locally:

```
$ python tests/test_new_tasks.py; beep3
[creating task(s): ms_marco:DefaultTeacher]
[building data: /private/home/roller/working/parlai/data/MS_MARCO]
[ downloading: https://msmarco.blob.core.windows.net/msmarco/train_v2.1.json.gz to /private/home/roller/working/parlai/data/MS_MARCO/train.gz ]
Downloading train.gz: 0.00B [00:00, ?B/s]
[ downloading: https://msmarco.blob.core.windows.net/msmarco/dev_v2.1.json.gz to /private/home/roller/working/parlai/data/MS_MARCO/valid.gz ]
Downloading valid.gz: 0.00B [00:00, ?B/s]
[ downloading: https://msmarco.blob.core.windows.net/msmarco/eval_v2.1_public.json.gz to /private/home/roller/working/parlai/data/MS_MARCO/test.gz ]
Downloading test.gz: 0.00B [00:00, ?B/s]
building fbformat:train
Reading gzip file /private/home/roller/working/parlai/data/MS_MARCO/train.gz.
Converting: 100%|██████████████████████████████████████████| 808731/808731 [00:02<00:00, 275386.15it/s]
building fbformat:valid
Reading gzip file /private/home/roller/working/parlai/data/MS_MARCO/valid.gz.
Converting: 100%|██████████████████████████████████████████| 101093/101093 [00:00<00:00, 202522.93it/s]
building fbformat:test
Reading gzip file /private/home/roller/working/parlai/data/MS_MARCO/test.gz.
Converting: 100%|██████████████████████████████████████████| 101092/101092 [00:00<00:00, 258962.59it/s]
[ loaded 808731 episodes with a total of 808731 examples ]
[creating task(s): ms_marco:PassageTeacher]
[loading fbdialog data:/private/home/roller/working/parlai/data/MS_MARCO/train.passage.txt]
[loading fbdialog data:/private/home/roller/working/parlai/data/MS_MARCO/train.passage.txt]
[ loaded 502939 episodes with a total of 502939 examples ]
[creating task(s): ms_marco:DefaultTeacher]
[ loaded 808731 episodes with a total of 808731 examples ]
[creating task(s): ms_marco:PassageTeacher]
[loading fbdialog data:/private/home/roller/working/parlai/data/MS_MARCO/train.passage.txt]
[loading fbdialog data:/private/home/roller/working/parlai/data/MS_MARCO/train.passage.txt]
[ loaded 502939 episodes with a total of 502939 examples ]
.
----------------------------------------------------------------------
Ran 1 test in 1252.512s

OK
python tests/test_new_tasks.py  1219.23s user 33.12s system 99% cpu 20:56.17 total
```

**Comment by stephenroller** [comment]
I wrapped up the final bits. There were some edge cases in the data set that made it ill suited for parlai format but I think I worked out all the kinks.

Thanks for your contribution @isaacmg.
